src/main/kotlin/com/londogard/textgen/tokenizers/SimpleExtensibleTokenizer.kt

```diff
@@ -0,0 +1,83 @@
+package com.londogard.textgen.tokenizers
+
+import java.util.*
+import java.util.regex.Pattern
+
+/**
+ * A simple tokenizer which allows you to define your own whitespace to split upon.
+ */
+class SimpleExtensibleTokenizer(
+    private val splitContraction: Boolean = false,
+    private val whitespace: Pattern = WHITESPACE
+    ) : Tokenizer {
+    override fun split(text: String): Array<String> {
+        var updatedText = text
```

Comment:
What is the reason for using mutability?

Reply:
This is not actually mutability but rather changing the reference. 
Basically this is a copy of another Tokenizer with some changes. 

Also, it's annoying to create 500 variable names xD